server/src/main/java/org/elasticsearch/index/mapper/SourceValueFetcher.java

```diff
@@ -85,6 +86,17 @@ public SourceValueFetcher(String fieldName, MapperService mapperService, boolean
         return values;
     }
 
+    private Object tryParseSourceValue(Object value) {
+        try {
+            return parseSourceValue(value);
+        } catch (IllegalArgumentException | ElasticsearchParseException e) {
+            // Skip any values that couldn't be parsed instead of failing the entire search
+            // request. It's possible to have malformed values if a field was previously unmapped
+            // and only present in the source, then later a mapping was added for the field.
+            return null;
```

Comment:
Should these be added to the list of ignored fields somehow? I've never been 100% clear on how that works, but this feels like a good candidate for it.

Reply:
I feel bad just returning null. Maybe you could log this at `debug`? I feel like one day I'm going to need to be able to check to see if it is happening and I'd regret it if I couldn't get the exception.